include/PriorityQueue.h

```diff
@@ -9,8 +9,32 @@ class PriorityQueue {
 
 public:
     void push(const T &t);
+
     T poll();
 
 private:
+    struct {
```

Comment:
Declaring structs this way is a bit old-fashion. In modern C++ it is better to declare the struct separately and then use it as a type. 
In this case you would declare the `Comparator` struct and then set `Comparator m_comp` as a member